src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Diagnostics/RazorDiagnosticsPublisher.cs

```diff
@@ -172,21 +183,47 @@ internal void ClearClosedDocuments()
     internal async Task PublishDiagnosticsAsync(IDocumentSnapshot document)
     {
         var result = await document.GetGeneratedOutputAsync();
+        var uriBuilder = new UriBuilder()
+        {
+            Scheme = Uri.UriSchemeFile,
+            Path = document.FilePath,
+            Host = string.Empty,
+        };
+        
+        var delegatedParams = new DocumentDiagnosticParams
+        {
+            TextDocument = new TextDocumentIdentifier { Uri = uriBuilder.Uri },
+        };
 
-        var diagnostics = result.GetCSharpDocument().Diagnostics;
+        var delegatedResponse = await _languageServer.SendRequestAsync<DocumentDiagnosticParams, SumType<FullDocumentDiagnosticReport, UnchangedDocumentDiagnosticReport>?>(
+            RazorLanguageServerCustomMessageTargets.RazorPullDiagnosticEndpointName,
+            delegatedParams,
+            CancellationToken.None).ConfigureAwait(false);
 
-        lock (PublishedDiagnostics)
+        var razorDiagnostics = result.GetCSharpDocument().Diagnostics;
+        IReadOnlyList<Diagnostic>? csharpDiagnostics = null;
+
+        lock (PublishedRazorDiagnostics)
+        lock (PublishedCSharpDiagnostics)
         {
             var filePath = document.FilePath.AssumeNotNull();
+            if (delegatedResponse != null && delegatedResponse.Value.First.Kind == "full")
```

Comment:
Could we potentially make `full` into a constant or is there an existing constant for it that we can use?